connector_woocommerce/models/product_product/export_mapper.py

```diff
@@ -118,7 +118,9 @@ def parent_id(self, record):
     def image(self, record):
         # WooCommerce only allows one image per variant product
         if (
-            record.product_image_attachment_ids
+            record.with_context(
+                include_main_product_image=self.backend_record.use_main_product_image
+            ).product_image_attachment_ids
             and self.backend_record.wordpress_backend_id
```

Comment:
```suggestion
        product_image_attachments = record.with_context(
                include_main_product_image=self.backend_record.use_main_product_image
        ).product_image_attachment_ids
        if (
            product_image_attachments
            and self.backend_record.wordpress_backend_id
```
and replace `image = record.product_image_attachment_ids[0].attachment_id` below with `image = product_image_attachments[0].attachment_id`
otherwise you'll still have all the images included the main one.